### PR TITLE
fix plugin init

### PR DESCRIPTION
**Pull request by madhur-tandon**
CC: @fpliger when discussing plugins for vizzu...

### CODE REVIEW

**Review by tedpatrick — APPROVED** [review]

File: pyscriptjs/tests/py-unit/test_python_plugins.py [hunk 1]
```diff
@@ -130,7 +132,7 @@ def test_connected_calls(self, plugins_manager: ppt.PluginsManager):
         # - init the plugin instance passing the plugins_manager as parent app
         # - add the plugin instance to plugins_manager.plugins
         assert not py_tutor.plugin.app
-        plugins_manager.addPythonPlugin(py_tutor.plugin)
+        await plugins_manager.addPythonPlugin(py_tutor.plugin)
 
         # EXPECT: the plugin app to now be the plugin manager
         assert py_tutor.plugin.app == plugins_manager
```
Comment (fpliger) [inline pyscriptjs/tests/py-unit/test_python_plugins.py:138]: How was this passing before? Did you actually test the failure mode and the it passes after you fixed the issue?

File: pyscriptjs/tests/py-unit/test_python_plugins.py [hunk 2]
```diff
@@ -161,7 +164,7 @@ def test_plugin_registered(self, plugins_manager: ppt.PluginsManager):
         assert not plugins_manager.plugins
 
         # GIVEN THAT we add the plugin to the app plugin manager
-        plugins_manager.addPythonPlugin(py_tutor.plugin)
+        await plugins_manager.addPythonPlugin(py_tutor.plugin)
```
Comment (fpliger) [inline pyscriptjs/tests/py-unit/test_python_plugins.py:167]: Feels like we need to add a check for  what was fixed

**Review by fpliger — CHANGES_REQUESTED** [review]
I think it's important that we add a test to verify that we actually fixed the issue.... Can you add that @madhur-tandon ?

File: pyscriptjs/src/python/pyscript/_plugin.py [hunk 3]
```diff
@@ -12,7 +12,7 @@ def __init__(self, name=None):
         self._custom_elements = []
         self.app = None
 
-    def init(self, app):
+    async def init(self, app):
```
Comment (JeffersGlass) [inline pyscriptjs/src/python/pyscript/_plugin.py:15]: Is `init` part of the public API for a Python plugin, i.e. are we encouraging users to override/extend it? If so, we should document this change.

**Pull request closed without merging**